### PR TITLE
Move 'properly validated keys' guidance before git verification...

### PR DESCRIPTION
**Pull request by dksmiffs**
...in the "How to Verify Qubes Repos" section, since you must have
properly validated keys before being able to perform a successful
`git verify-tag` or `git verify-commit`.

### CODE REVIEW

File: project-security/verifying-signatures.md [hunk 1]
```diff
@@ -507,6 +507,19 @@ signed tags or commits on top of them unless you personally vouch for the
 trustworthiness of the unsigned commits. Instead, ask the person who pushed the
 unsigned commits to sign them.
 
+You should always perform this verification on a trusted local machine with
+properly validated keys (which are available in the [Qubes Security
+Pack](/security/pack/)) rather than relying on a third party, such as GitHub.
```
Comment (unman) [inline project-security/verifying-signatures.md:512]: _this verification_ has now lost it's referent

**Review by andrewdavidwong — COMMENTED** [review]
> ...in the "How to Verify Qubes Repos" section, since you must have
> properly validated keys before being able to perform a successful
> `git verify-tag` or `git verify-commit`.

If you look at where this section appears in the context of the whole page, how to obtain and authenticate keys was already covered above. The section you're moving isn't intended to be instructions for how to obtain and properly validate keys, since that has its own section.

**Comment by andrewdavidwong** [comment]
> > ...in the "How to Verify Qubes Repos" section, since you must have
> 
> > properly validated keys before being able to perform a successful
> 
> > `git verify-tag` or `git verify-commit`.
> 
> 
> 
> If you look at where this section appears in the context of the whole page, how to obtain and authenticate keys was already covered above. The section you're moving isn't intended to be instructions for how to obtain and properly validate keys, since that has its own section.

Oh, sorry, I was thinking of the wrong keys. The instructions for obtaining and authenticating the keys relevant to this actually are on the secpack page, so it makes sense to link there earlier.

File: project-security/verifying-signatures.md (hunk 1 above)
Comment (dksmiffs) [reply project-security/verifying-signatures.md:512]: I changed back to "this verification".

File: project-security/verifying-signatures.md [hunk 2]
```diff
@@ -498,15 +498,28 @@ can be confident that these hash values came from the Qubes devs.
 ## How to Verify Qubes Repos
 
 Whenever you use one of the [Qubes repositories](https://github.com/QubesOS),
-you should verify the PGP signature in a tag on the latest commit or on the
-latest commit itself. (One or both may be present, but only one is required.)
-If there is no trusted signed tag or commit on top, any commits after the
-latest trusted signed tag or commit should **not** be trusted. If you come
+you should use Git to verify the PGP signature in a tag on the latest commit or
+on the latest commit itself (one or both may be present, but only one is
+required.) If there is no trusted signed tag or commit on top, any commits after
```
[the remark this answers is not in the document]
Comment (dksmiffs) [reply project-security/verifying-signatures.md]: Returned to grammatically correct wording.

**Comment by dksmiffs** [comment]
> If you look at where this section appears in the context of the whole page, how to obtain and authenticate keys was already covered above. The section you're moving isn't intended to be instructions for how to obtain and properly validate keys, since that has its own section.

Okay, but I'm arriving at this section not from a whole page perspective, but rather via the Qubes FAQ, [Should I trust this website?](https://www.qubes-os.org/faq/#should-i-trust-this-website):

"Instead, if you want to obtain your own trustworthy copy of this website in a secure way, you should"
  1. "clone our website repo"
  2. "verify the PGP signatures on the commits and/or tags"

And that last step links directly to this "How to Verify Qubes Repos" section.  So via that path, the reader *is* seeking instructions for how to obtain and properly validate keys.

**Comment by andrewdavidwong** [comment]
> > If you look at where this section appears in the context of the whole page, how to obtain and authenticate keys was already covered above. The section you're moving isn't intended to be instructions for how to obtain and properly validate keys, since that has its own section.
> 

Please see my subsequent comment rescinding this. It was my mistake.

> 
> Okay, but I arrived at this section not from a whole page perspective, but rather via the Qubes FAQ, [Should I trust this website?](https://www.qubes-os.org/faq/#should-i-trust-this-website):
> 
> 
> 
> "Instead, if you want to obtain your own trustworthy copy of this website in a secure way, you should"
> 
>   1. "clone our website repo"
> 
>   2. "verify the PGP signatures on the commits and/or tags"
> 
> 
> 
> And that last step links directly to this "How to Verify Qubes Repos" section.  So via that path, the reader *is* seeking instructions for how to obtain and properly validate keys.
> 

Right, this is why I agree with you that it's important to link to the secpack page early, since it contains those instructions. 

> 
> On a review of the whole page, I think the awkwardness stems from the fact that all three of the major "How to" sections on this page have properly validated keys as a prerequisite.  Should we adjust the overall page structure to something like the following, which would allow shortening the existing "How to" sections?
> 

Yes, but as mentioned above, the key instructions for the last section are actually on the secpack page, not this one. 

The instructions earlier on this page are only for the QMSK and release keys, which won't help with verifying sigs on Git objects.

> 
> - What Digital Signatures Can and Cannot Prove
> 
> - Prerequisite: Obtain and Properly Validate Keys
> 
> - How to Verify Qubes ISO Signatures
> 
> - How to Verify Qubes ISO Digests
> 
> - How to Verify Qubes Repos
> 
> - Troubleshooting FAQ
> 
> 

I agree with the idea. Then we would link to the prereq section near the beginning of each of the next two sections (for people who arrive via anchor links, like you did).

**Pull request merged**